WordPress/openverse
# Add sensitivity toggles in filter sidebar and modal

## Fixes



Fixes #2368 by @dhruvkb

## Description



This PR adds toggles in the sidebar to control the fetching and blurring of sensitive content.

The next step for the project would be to link the first toggle with the search store's filter mechanism so that changing it applies the right query params in the search request, i.e. #2548

## Testing Instructions


0. Check out the PR and run the dev server.
1. Enable `fake_sensitive` and `sensitive_content` flags. The former ensures there is some content to test blurring with and the latter controls the visibility of the toggles.
2. Search for something.
3. Open the filter sidebar, you should see the toggles.
4. The first toggle controls the `fetch_sensitive` feature flag. Toggle it to see the session storage value change.
5. The second toggle controls whether sensitive content appears blurred. Toggle it to blur/unblur the search results.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
